drivers/disk/usdhc.c

```diff
@@ -2596,6 +2611,34 @@ static int usdhc_sd_init(struct usdhc_priv *priv)
 
 static K_MUTEX_DEFINE(z_usdhc_init_lock);
 
+
+static int usdhc_dat3_pull(bool pullup, struct usdhc_priv *priv)
+{
+	int ret = 0U;
+
+	if (pullup) {
+		/* Just call the board specific function (registered in SOC file) */
+		imxrt_usdhc_dat3_pull(pullup);
+	} else {
+		/* Call board specific function to pull down DAT3 */
+		imxrt_usdhc_dat3_pull(pullup);
+		/* Power off the card to clear DAT3 legacy status */
```

Comment:
doing the same function call at beginning of each sides of if statement. Why not pull the function call out and just do a 

```
imxrt_usdhc_dat3_pull(pullup);

if (!pullup) {
...
}
```

Reply:
Fixed in latest push